File: TellDontAsk/AlarmsObjectOriented/Criterias/NightCriteria.cs
using System;
using AlarmsObjectOriented.Interfaces;

namespace AlarmsObjectOriented.Criterias
{
  internal class NightCriteria : TimeCriteria
  {
    public bool IsSatisfied()
    {
      return !new DayCriteria().IsSatisfied();
    }

    public void Output()
    {
      Console.WriteLine("it's night");
    }
  }
}

File: Java/AlarmsObjectOriented/src/main/java/criterias/NightCriteria.java
package criterias;

import interfaces.TimeCriteria;

public class NightCriteria implements TimeCriteria {
  public boolean isSatisfied() {
    return !new DayCriteria().isSatisfied();
  }

  public void output() {
    System.out.println("it's night");
  }
}
